### PR TITLE
feat: add securityContext and podSecurityContext options

### PR DESCRIPTION
**Pull request by roobre**
Postfix and Dovecot sometimes need capabilities that some CRIs, such as cri-o, do not support by default, e.g. `SYS_CHROOT`. In these cases, if these capabilities are not added explicitly, postfix and dovecot won't start.

This commit adds `.Values.{postfix,dovecot}.containerSecurityContext` to values.yaml and to the corresponding templates.

This PR is effectively the same as https://github.com/Mailu/helm-charts/pull/125, which seems to have been lost in the 1.x migration.

### CODE REVIEW

File: mailu/templates/dovecot/deployment.yaml [hunk 1]
```diff
@@ -65,6 +65,10 @@ spec:
         - name: dovecot
           image: {{ .Values.imageRegistry }}/{{ .Values.dovecot.image.repository }}:{{ default (include "mailu.version" .) .Values.dovecot.image.tag }}
           imagePullPolicy: {{ .Values.dovecot.image.pullPolicy }}
+          {{- with .Values.dovecot.containerSecurityContext }}
```
Comment (kang-makes) [inline mailu/templates/dovecot/deployment.yaml:68]: The idiomatic way in Helm us to call `securityContext` to container's security policy and `podSecurityContext` to pod's security policy.

**Review by kang-makes — COMMENTED** [review]
Y would add `podSecurityPolicy in this PR too. We are not affected in our cluster but people that can potentially use `securityPolicy` to run things as non root or drop privileges would also need `fsGroup` at pod's security context level.

File: mailu/templates/dovecot/deployment.yaml (hunk 1 above)
Comment (roobre) [reply mailu/templates/dovecot/deployment.yaml]: Done, thanks for the feedback!

File: mailu/values.yaml [hunk 2]
```diff
@@ -1240,6 +1240,11 @@ postfix:
   ##     my_variable = my_value
   overrides: {}
 
+  ## @param postfix.securityContext `securityContext` for postfix container
+  securityContext: {}
+  ## @param postfix.securityContext `securityContext` for postfix pod
```
Comment (fastlorenzo) [inline mailu/values.yaml:1245]: replace `postfix.securityContext` with `postfix.podSecurityContext`

**Review by fastlorenzo — CHANGES_REQUESTED** [review]
Thanks for the PR, that's a good idea!
Could you also make it for other services to keep it consistent? 
Thanks!

File: mailu/values.yaml (hunk 2 above)
Comment (roobre) [reply mailu/values.yaml]: Good catch, should be fixed now!
I've also rebased to latest `master`.

**Comment by fastlorenzo** [comment]
@roobre I've made some changes to support `podSecurityContext` and `containerSecurityContext` for all services (not just dovecot and postfix).
Could you try this PR before I merge it?

**Comment by fastlorenzo** [comment]
fixes #293

**Review by fastlorenzo — APPROVED** [review]

**Comment by roobre** [comment]
Hey @fastlorenzo, apologies for the delay getting back to you.

I'm testing this right now, and unfortunately the latest changes are not working for me.

If I try to add the capabilities `CRI-O` requires by specifying:
```yaml
  postfix:
    containerSecurityContext:
      enabled: true
      capabilities:
        add:
          - SYS_CHROOT
```

This will unfortunately also bring:

```yaml
    runAsUser: 1001
    runAsNonRoot: false
```

Which is not something I want in the deployment. To get rid of them, I'd need to manually override those. While I can do that, I don't think that's ideal as they may get out of sync with the chart.

**Pull request merged**